### PR TITLE
Waterfall callback

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: xpdview/waterfall.py [hunk 1]
```diff
@@ -107,8 +106,8 @@ def _adapt_data_list(self, int_data_list):
     def on_plot_hover(self, event):
         """callback to show legend when click on one of curves"""
         line = event.artist
-        name = line.get_label()
-        line.axes.legend([name],handlelength=0,
+        name = line.get_gid()
+        line.axes.legend([name], handlelength=0,
                          handletextpad=0, fancybox=True)
```
Comment (chiahaoliu) [inline xpdview/waterfall.py]: I would suggest to use ``get_label`` unless special purpose. ``label`` is the natural mpl API

**Comment by CJ-Wright** [comment]
@chiahaoliu ready for more review

File: xpdview/waterfall.py [hunk 2]
```diff
@@ -122,13 +126,13 @@ def _update_plot(self, x_offset_val=None, y_offset_val=None):
...
-        x_array_list, y_array_list,\
-                y_min, y_max, y_dist, x_min, x_max, x_dist = state
+        x_array_list, y_array_list, \
+        y_min, y_max, y_dist, x_min, x_max, x_dist = state
         for ind, el in enumerate(zip(x_array_list, y_array_list)):
             x, y = el
             self.ax.plot(x + x_dist * ind * x_offset_val,
                          y + y_dist * ind * y_offset_val,
-                         label=self.key_list[ind], picker=5)
+                         gid=self.key_list[ind], picker=5)
```
Comment (chiahaoliu) [inline xpdview/waterfall.py]: need to replace ``gid`` with ``label``

**Comment by CJ-Wright** [comment]
@chiahaoliu ready for round three

File: xpdview/callbacks.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from bluesky.callbacks.broker import BrokerCallbackBase
+from .waterfall import Waterfall
+import matplotlib.pyplot as plt
+
+
+class LiveWaterfall(BrokerCallbackBase):
+    """
+    Stream 2D images in a cross-section viewer.
```
Comment (chiahaoliu) [inline xpdview/callbacks.py]: should it be 1D ?

File: xpdview/callbacks.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    """
+
+    def __init__(self, x_name, y_name, db=None, units=None):
+        import matplotlib.pyplot as plt
+        super().__init__((x_name, y_name,), db=db)
+        self.db = db
+        self.x_name = x_name
+        self.y_name = y_name
+        self.units = units
+        self.fig = plt.figure()
```
Comment (chiahaoliu) [inline xpdview/callbacks.py]: suggest to use 
````
from matplotlib.figure import Figure
self. fig = Figure()
````
``plt`` comes with its own complex backends to identify the states which sometimes mess things up.

Wisdom from @tacaswell

**Review by chiahaoliu — COMMENTED** [review]
LGTM except for the minor comments

**Pull request merged**